src/main/scala/interfaces/Converter.scala

```diff
@@ -0,0 +1,318 @@
+            case Null() => return model.entries("$Ref.null").toString
+            case Var(id, sort) => {
+                val key = term.toString
+                //this can fail : TODO throw and catch exception
+                return model.entries(key).toString
+            }
+            case t@App(app, args) => {
+                println("Found type APP")
+                /* not tested yet, not sure for which examples this occurs on heap*/
+                var fname = app.id + "%limited"
+                if (!model.entries.contains(fname)){
+                    fname = app.id.toString
+                    if (!model.entries.contains(fname)){
+                        fname = fname.replace("[","<").replace("]",">")
+                    }
+                }
+                var arg = snapToOneLine(args.map(x => evaluateTerm(x, model)).mkString(" "))
+                return getFunctionValue(model, fname, arg)
+            }
+            case Combine(p0, p1) => {
+                val p0eval = evaluateTerm(p0, model)
+                val p1eval = evaluateTerm(p1, model)
+                return "($Snap.combine " + p0eval + " " + p1eval + ")"
+            }
+            case First(p) => {
+                val sub = evaluateTerm(p, model)
+                if (sub.startsWith("($Snap.combine")){
+                    return getParts(sub)(1)
+                }
+                println("WARNING: one heap entry could not be resolved")
+                return ""
+            }
+            case Second(p) => {
+                val sub = evaluateTerm(p, model)
+                if (sub.startsWith("($Snap.combine")){
+                    return getParts(sub)(2)
+                }
+                println("WARNING: one heap entry could not be resolved")
+                return ""
+            }
+            case SortWrapper(t, to) =>  {
+                val sub = snapToOneLine(evaluateTerm(t, model))
+                val fromSortName : String = translateSort(t.sort)
+                val toSortName : String = translateSort(to)
+                val fname = "$SortWrappers." + fromSortName + "To" + toSortName
+                return getFunctionValue(model, fname, sub)
+            }
+            case PredicateLookup(predname, psf, args) => {
+                /* not tested! did never occurr in considered examples */
+                val lookupFuncName : String = "$PSF.lookup_" + predname
+                val snap = toSnapTree.apply(args)
+                val psfVal = evaluateTerm(psf, model)
+                val snapVal = evaluateTerm(snap, model)
+                val arg = snapToOneLine(psfVal + " " + snapVal)
+                return getFunctionValue(model, lookupFuncName, arg)
+            }
+            case _ =>   println("of unhandled type") 
+                        return ""
+        }
+    }
+
+    def extractHeap(h:Iterable[Chunk], model: Model) : ExtractedHeap = {
+        var target : ExtractedHeap = Map()
+        for (chunk <- h) {
```

Comment:
Style: lines 202 to 204 should be equivalent to

    h foreach {
      case c@BasicChunk ...